src/engine/CartesianProductJoin.cpp

```diff
@@ -260,7 +260,9 @@ CartesianProductJoin::calculateSubResults(bool requestLaziness) {
   // Get all child results (possibly with limit, see above).
   for (std::shared_ptr<QueryExecutionTree>& childTree : children_) {
     if (limitIfPresent.has_value() && childTree->supportsLimit()) {
-      childTree->setLimit(limitIfPresent.value());
+      // NOTE: If computeResult is called more than once on the same instance of
+      // `CartesianProduct`
```

Comment:
.... then what?

Reply:
And doesn't this highly speak against applying this limit at this place etc, or detecting this case and throwing etc.?